src/data/polynomial/hasse_deriv.lean

```diff
@@ -162,6 +172,41 @@ begin
   all_goals { apply_rules [mul_ne_zero, H] }
 end
 
+lemma nat_degree_hasse_deriv_le (p : polynomial R) (n : ℕ) :
+  nat_degree (hasse_deriv n p) ≤ nat_degree p - n :=
+begin
+  classical,
+  rw [hasse_deriv_apply, sum_def],
+  refine (nat_degree_sum_le _ _).trans _,
+  simp_rw [function.comp, nat_degree_monomial],
+  rw [finset.fold_ite, finset.fold_const],
+  { simp only [if_t_t, max_eq_right, zero_le', finset.fold_max_le, true_and, and_imp,
+               tsub_le_iff_right, mem_support_iff, ne.def, finset.mem_filter],
+    intros x hx hx',
+    have hxp : x ≤ p.nat_degree := le_nat_degree_of_ne_zero hx,
+    have hxn : n ≤ x,
+    { contrapose! hx',
+      simp [nat.choose_eq_zero_of_lt hx'] },
+    rwa [tsub_add_cancel_of_le (hxn.trans hxp)] },
+  { simp }
```

Comment:
This lemma could probably proved in a way similar to the non-`le` version, but maybe it is not worth the trouble (and could certainly be left for a future PR).